src/pudl/metadata/classes.py

```diff
@@ -1648,8 +1648,10 @@ def enforce_schema(self, df: pd.DataFrame) -> pd.DataFrame:
             raise ValueError(
                 f"{self.name} {len(dupes)}/{len(df)} duplicate primary keys ({pk=}) when enforcing schema."
             )
-        if pk and df.loc[:, pk].isna().any(axis=None):
-            raise ValueError(f"{self.name} Null values found in primary key columns.")
+        if pk and not (nulls := df[df[pk].isna().any(axis=1)]).empty:
+            raise ValueError(
+                f"{self.name} Null values found in primary key columns.\n{nulls}"
+            )
```

Comment:
just a minor edit so you can see the IDs that are missing in the logs

Reply:
Great, this is a super useful log addition.